### DISCUSSION TITLE
esp32-s2 arduino library

### DISCUSSION THREAD

**Discussion by chegewara** [hathach/tinyusb#583] [Show and tell]
I would like to thank you for this library with very easy to understand examples. Thanks for that i was able to make simple and easy to use arduino library that is designed for esp32-s2 and probably S3 too, when it will be released.
Library is designed to create devices with few basic usb classess:
- CDC-ACM,
- MSC,
- MIDI,
- DFU,
- webUSB,
- HID: keyboard, mouse, gamepad, generic IN/OUT.

Maybe its not best designed library, but the idea was to make it as simple to use as possible, since its arduino library. 

One example is using m5stack cardKB, its credit card size keyboard with I2C interface.

**Comment by hathach** [comment]
Superb !!! thank you letting me know, that is really a great library. I am sure lots of people will find it userful. Adafruit also has an platform-indepdence Arduino library for TinyUSB as well https://github.com/adafruit/Adafruit_TinyUSB_Arduino. It currently works on both SAMD and nRF, part of my job is to have it running on esp32s2 as well, last time I check espressif is also doing some usb-related work for Arduino library also. The more choice the better.
